pkg/util/adoption/conflictresolution_test.go

```diff
@@ -0,0 +1,129 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package adoption
+
+import (
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func newDeployment(annotations map[string]string) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "test",
+			Annotations: annotations,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "test",
+							Image: "test",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func TestFilterToAdoptCluster(t *testing.T) {
+	type args struct {
+		obj         metav1.Object
+		clusterName string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "propagate resource without adopting",
+			args: args{
+				obj:         newDeployment(nil),
+				clusterName: "kubeadmiral-member-1",
+			},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "invalid annotation for clusters to adopt",
+			args: args{
+				obj:         newDeployment(map[string]string{ClustersToAdoptAnnotation: "annotation"}),
+				clusterName: "kubeadmiral-member-1",
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "cluster not matched",
+			args: args{
+				obj:         newDeployment(map[string]string{ClustersToAdoptAnnotation: `{"clusters": ["kubeadmiral-member-1"]}`}),
+				clusterName: "kubeadmiral-member-2",
+			},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "cluster name matched",
+			args: args{
+				obj:         newDeployment(map[string]string{ClustersToAdoptAnnotation: `{"clusters": ["kubeadmiral-member-1"]}`}),
+				clusterName: "kubeadmiral-member-1",
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "invalid regexp",
+			args: args{
+				obj:         newDeployment(map[string]string{ClustersToAdoptAnnotation: `{"regexp": "*"}`}),
+				clusterName: "kubeadmiral-member-1",
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "regexp matched",
+			args: args{
+				obj:         newDeployment(map[string]string{ClustersToAdoptAnnotation: `{"regexp": ".*"}`}),
+				clusterName: "kubeadmiral-member-1",
+			},
+			want:    true,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := FilterToAdoptCluster(tt.args.obj, tt.args.clusterName)
```

Comment:
`FilterToAdoptCluster` takes in a `FederatedObject`. Users usually refer to tests to study the intended usage of functions. Is it possible to change `newDeployment` to `newFederatedObject` to prevent confusion?